src/Tests/Microsoft.NET.Build.Tests/AppHostTests.cs

```diff
@@ -16,17 +17,45 @@
 using Microsoft.NET.TestFramework.ProjectConstruction;
 using Xunit;
 using Xunit.Abstractions;
+using NuGet.Frameworks;
 
 namespace Microsoft.NET.Build.Tests
 {
     public class AppHostTests : SdkTest
     {
+        private static string[] GetExpectedFilesFromBuild(TestAsset testAsset, string targetFramework)
+        {
+            var testProjectName = testAsset.TestProject?.Name ?? testAsset.Name;
+            var expectedFiles = new List<string>()
+            {
+                $"{testProjectName}{Constants.ExeSuffix}",
+                $"{testProjectName}.dll",
+                $"{testProjectName}.pdb",
+                $"{testProjectName}.deps.json",
+                $"{testProjectName}.runtimeconfig.json"
+            };
+
+            if (!string.IsNullOrEmpty(targetFramework))
+            {
+                var parsedTargetFramework = NuGetFramework.Parse(targetFramework);
+                if (parsedTargetFramework.Version >= new Version(5, 0, 0, 0))
```

Comment:
The `Version` type has somewhat unexpected comparison rules (5.0 is not the same as 5.0.0.0), so I would suggest just comparing against the `Major` version.
```suggestion
                if (parsedTargetFramework.Version.Major >= 5)
```